### DISCUSSION TITLE
Keep Strapi-Plugin models in sync with UI & DB between plugin updates.

### DISCUSSION THREAD

**Discussion by ivanjeremic** [strapi/strapi#7269] [Legacy Topics]
**THE ISSUE**

While developing my first plugin for Strapi I came across one issue related to models and keeping them updated in a newer version of the plugin.

My plugin generates Collection-Types with all attributes needed on the first startup and this works very well, but the issue I have, we as plugin developers also need to ship updates from time to time and what if I need to modify my models and ship new fields or update/delete them?

I'm not sure if there are many use cases for this but it seems to me there are. I think it is worth a discussion because if the plugin ecosystem should become the key feature of Strapi I think some other developers would love to extending models when shipping updates and be confident that it will be represented correctly in the plugin users user interface and also when editing entries be confident the newly added fields are also updated in the database.

**EDIT**
It seems to add additional attributes/fields in updates works fine, but I tested it only locally with the quickstart installation.

**Comment by derrickmehaffy** [comment]
@soupette @alexandrebodin I think this plays into our discussion with the new plugin API.
